### PR TITLE
[Document Provider] Create Files

### PR DESCRIPTION
**Pull request by abelgardep**
Implements #2539

### CODE REVIEW

File: owncloudApp/src/main/java/com/owncloud/android/providers/DocumentsStorageProvider.kt [hunk 1]
```diff
@@ -65,66 +66,84 @@ class DocumentsStorageProvider : DocumentsProvider() {
...
+                    if (!waitOrGetCancelled(signal)) {
+                        return null
+                    }
+                    ocFile = currentStorageManager?.getFileById(docId)
+                        ?: throw FileNotFoundException("Failed to open document with id $documentId and mode $mode")
+
+                } while (!ocFile.isDown)
+            }
+        } else {
+            ocFile = fileToUpload
```
Comment (davigonz) [inline owncloudApp/src/main/java/com/owncloud/android/providers/DocumentsStorageProvider.kt]: If we are able to know when the provider uses this method to upload or download a file, maybe we do not need to use SynchronizeFileOperation to perform a simple upload and a call to `uploadNewFile` of `TransferRequester.java` from here is enough. This way we reduce some logic in SynchronizeFileOperation by removing the `uploadOnly` parameter and do not overload it with extra responsibilities. What do you think?

You could create a transferRequest in this else condition

```
val transferRequester = TransferRequester()
transferRequester.uploadNewFile(context, currentStorageManager?.account, ..., ...)
```

Comment (abelgardep) [reply owncloudApp/src/main/java/com/owncloud/android/providers/DocumentsStorageProvider.kt]: Yes, that would be definitively better

Comment (davigonz) [reply owncloudApp/src/main/java/com/owncloud/android/providers/DocumentsStorageProvider.kt]: Great, ping me when is ready so I can continue with the review, thanks!

Comment (abelgardep) [reply owncloudApp/src/main/java/com/owncloud/android/providers/DocumentsStorageProvider.kt]: But we'll need to create that transfer requester later, as it is done at that commit 63e19bf97d6c11af1705597ecc7bc36b0e1a8226

**Comment by davigonz** [comment]
@jesmrec code approved, ready to test

**Comment by jesmrec** [comment]
Approved

**Pull request merged**